### DISCUSSION TITLE
Tracker showing duplicate torrents

### DISCUSSION THREAD

**Discussion by ianstalk** [haugene/docker-transmission-openvpn#1959] [Q&A] [answered]
I've been seeing an odd behavior where my private tracker is showing the same torrent multiple times from different IP addresses. Also sometimes torrents will stop seeding even though I can still access Transmission via RPC, I'm not sure if the two are related. I'm running Ubuntu 18.04.5 Bionic and Docker version 20.10.7. 

I'm starting the container as a systemd service using the following .service file:
```
[Unit]
Description=Transmission-OpenVPN Container
After=docker.service
Requires=docker.service

[Service]
User=transmission-openvpn
Group=docker
TimeoutStartSec=0
Restart=always
ExecStartPre=-/usr/bin/docker stop %n
ExecStartPre=-/usr/bin/docker rm %n
ExecStartPre=/usr/bin/docker pull haugene/transmission-openvpn
ExecStart=/usr/bin/docker run --rm --name %n --cap-add=NET_ADMIN -v /etc/transmission-openvpn:/data -v /mnt/drobo/downloads:/drobo --env-file /etc/transmission-openvpn/env.list --log-driver json-file --log-opt max-size=10m -p 9091:9091 haugene/transmission-openvpn

[Install]
WantedBy=multi-user.target
```

My docker .env file:
```
OPENVPN_PROVIDER=EXPRESSVPN
OPENVPN_CONFIG=my_expressvpn_usa_-_new_york_udp
OPENVPN_USERNAME=
OPENVPN_PASSWORD=
LOCAL_NETWORK=192.168.182.0/16
```

Any ideas?

**Comment by ianstalk** [comment]
Time will tell, but it seems it might be related to the multiple mount points.

**Reply by ianstalk** [reply]
Nope, it's happening again. No idea what's wrong.

**Reply by pkishino** [reply]
is your container restarting ? getting new IP assigned etc ? if it shows the same torrent from multiple IPs it could be that you IP is being re-assigned and tracker still showing old entries.. are you sure only single process running?

**Comment by ianstalk** [comment, marked as answer]
Hm, that might be it. I definitely still had remnants of an attempt to set
up openvpn on the host machine including both an init.d script and a
systemd service. Guess apt didn't do such a good job of cleanup. I've
removed those and am seeing fewer openvpn processes. I'll report back in a
few days if it's helped things. Thanks for the tip!

**Reply by pkishino** [reply]
ok, hopefully that is it

**Comment by ianstalk** [comment]
Hm, still seeing three separate IPs listed on my tracker. I only see
Transmission logs; is there any way for me to see if OpenVPN is crashing?

**Reply by pkishino** [reply]
check the container logs if it is restarting

**Reply by ianstalk** [reply]
I don't see anything that indicates a crash, but I do see the same few lines spammed 3 times every hour:

```
Sat Oct 16 21:04:51 2021 TLS: soft reset sec=0 bytes=354205/-1 pkts=2886/0
Sat Oct 16 21:04:51 2021 VERIFY OK: depth=1, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=ExpressVPN CA, emailAddress=support@expressvpn.com
Sat Oct 16 21:04:51 2021 VERIFY OK: nsCertType=SERVER
Sat Oct 16 21:04:51 2021 VERIFY X509NAME OK: C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-10265-0a, emailAddress=support@expressvpn.com
Sat Oct 16 21:04:51 2021 VERIFY OK: depth=0, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-10265-0a, emailAddress=support@expressvpn.com
Sat Oct 16 21:04:51 2021 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Sat Oct 16 21:04:51 2021 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Sat Oct 16 21:04:51 2021 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
Sat Oct 16 22:04:51 2021 TLS: soft reset sec=0 bytes=418385/-1 pkts=3674/0
Sat Oct 16 22:04:51 2021 VERIFY OK: depth=1, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=ExpressVPN CA, emailAddress=support@expressvpn.com
Sat Oct 16 22:04:51 2021 VERIFY OK: nsCertType=SERVER
Sat Oct 16 22:04:51 2021 VERIFY X509NAME OK: C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-10265-0a, emailAddress=support@expressvpn.com
Sat Oct 16 22:04:51 2021 VERIFY OK: depth=0, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-10265-0a, emailAddress=support@expressvpn.com
Sat Oct 16 22:04:51 2021 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Sat Oct 16 22:04:51 2021 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Sat Oct 16 22:04:51 2021 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
Sat Oct 16 23:04:51 2021 VERIFY OK: depth=1, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=ExpressVPN CA, emailAddress=support@expressvpn.com
Sat Oct 16 23:04:51 2021 VERIFY OK: nsCertType=SERVER
Sat Oct 16 23:04:51 2021 VERIFY X509NAME OK: C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-10265-0a, emailAddress=support@expressvpn.com
Sat Oct 16 23:04:51 2021 VERIFY OK: depth=0, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-10265-0a, emailAddress=support@expressvpn.com
Sat Oct 16 23:04:51 2021 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Sat Oct 16 23:04:51 2021 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Sat Oct 16 23:04:51 2021 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
```

**Comment by pkishino** [comment]
Yay, I see your provider is triggering soft-reset.. why I don’t know. You
might be best to ask their support.

**Comment by haugene** [comment]
Do you have logs over a larger period of time? If it soft-resets every hour it's typically the ping-check and inactivity stuff.
We're appending some config options in order to catch the case where the container looses connectivity.

This is done by the [modify-openvpn-config.sh](https://github.com/haugene/docker-transmission-openvpn/blob/master/openvpn/modify-openvpn-config.sh) script. You can try to disable just that part by setting environment variable `CONFIG_MOD_PING=0` and see how that goes :crossed_fingers:

**Reply by ianstalk** [reply]
Here's the full output of `docker logs` https://pastebin.com/AFsSajiu

**Reply by haugene** [reply]
As I expected. soft reset every hour. Try the config option I proposed. I think that will stop the container restarting. But it will also increase the chances of things just hanging if your connection goes stale. That just means that you have to restart the container yourself if it loses connectivity.
